### ISSUE TITLE
🐛 Panic `byte index 1 is not a char boundary` in `display/frame.rs`

### ISSUE THREAD

**Issue by qarmin** [biomejs/biome#4593]
### Environment information

```block
CLI:
  Version:                      0.0.0 (self compiled 6b183872744780ac55f7e6f45ac14f60b2796d49)
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           unset
  JS_RUNTIME_NAME:              unset
  NODE_PACKAGE_MANAGER:         unset

Biome Configuration:
  Status:                       unset

Workspace:
  Open Documents:               0
```

### What happened?

File content(at the bottom should be attached raw, not formatted file - github removes some non-printable characters, so copying from here may not work):
```
。<script>
{</script>
```

command
```
timeout -v 150 biome lint TEST___FILE.vue
```

cause this
```
/opt/BROKEN_FILES_DIR/newmessage_IDX_0_RAND_327260588916434309433686_minimized_824.vue:2:2 parse ━━━━━━━━━━

  × expected `}` but instead the file ends
  
    1 │ 。<script>
  > 2 │ {</script>
      │  
  
  i the file ends here
  
  > 1 │ 。<script>
      │ Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue:

Source Location: crates/biome_diagnostics/src/display/frame.rs:229:35
Thread Name: main
Message: byte index 1 is not a char boundary; it is inside '。' (bytes 0..3) of `。<script>`


##### Automatic Fuzzer note, output status "None", output signal "Some(6)"
```

### Expected result

No crash

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by qarmin** [comment]
I got same similar crash with typescript

File content(at the bottom should be attached raw, not formatted file - github removes some non-printable characters, so copying from here may not work):
```
/\
```

command
```
timeout -v 40 biome_normal lint TEST___FILE.ts
```

cause this
```
/opt/BROKEN_FILES_DIR/EXAMPLE_FILE_17730__IDX_0_RAND_417619510649650511328553_minimized_286.ts:1:2 lint/complexity/noUselessEscapeInRegex  FIXABLE  ━━━━━━━━━━

  i The character doesn't need to be escaped.
  
  > 1 │ /\
      │  Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue.

When opening the issue, please provide a minimal reproduction, or identify and share the file/code that triggers it. Without a way to reproduce the error, the error can't be fixed:

Source Location: crates/biome_diagnostics/src/display/frame.rs:240:57
Thread Name: main
Message: byte index 3 is out of bounds of `/\`
```

[compressed.zip](https://github.com/user-attachments/files/23644267/compressed.zip)

**Comment by arendjr** [comment]
Seeing how we have rolled out improved support for Vue, Svelte and Astro now, I think this one is no longer applicable. I'll optimistically close this one, but if there are similar issues remaining, we always appreciate your reports @qarmin !
